scripts/giphy.coffee

```diff
@@ -0,0 +1,27 @@
+# Commands:
+#   giphy <term> - Returns a randomly selected gif from a search of the giphy api for <term>
+    
+giphy =
+  api_key: process.env.HUBOT_GIPHY_API_KEY
+  api_url: 'http://api.giphy.com/v1'
+    
+  
+  search: (msg, q, callback) ->
+    endpoint = '/gifs/search'
+    url = "#{giphy.api_url}#{endpoint}"
+    msg.http(url)
+      .query
+        api_key: giphy.api_key
+        q: q
+      .get() (err, res, body) ->
+        res = JSON.parse(body)
+        data = res?.data || []
```

Comment:
in coffee, you can be slightly more semantically correct by using:

```coffeescript
data = res?.data ? []
```

See also http://coffeescript.org/#operators-existential